compress.go

```diff
@@ -22,6 +22,87 @@ var (
 			return gzip.NewWriter(nil)
 		},
 	}
+	gzipWriterPoolForCompressionLevel1 = sync.Pool{
+		New: func() interface{} {
+			gz, err := gzip.NewWriterLevel(nil, 1)
+			if err != nil {
+				panic(err)
```

Comment:
Not a big fan of panic, what happens if we reach here and panic?

Reply:
NewWriterLevel is like NewWriter but specifies the compression level instead of assuming DefaultCompression.

The compression level can be DefaultCompression, NoCompression, HuffmanOnly or any integer value between BestSpeed and BestCompression inclusive. **The error returned will be nil if the level is valid.**

1-9 level is correct.

Reply:
I am really not a big fan of using panic. but 👍